nsis/pcsx2_full_install.nsi

```diff
@@ -45,6 +53,69 @@ ShowUninstDetails nevershow
 ; =======================================================================
 
 ; -----------------------------------------------------------------------
+
+; The "" makes the section hidden.
+Section "" SecUninstallPrevious
+
+	Call UninstallPrevious
+	
+SectionEnd
+
+Function UninstallPrevious
+	;Check for uninstaller.
+	ReadRegStr $R0 HKLM "${INSTDIR_REG_KEY}" "UninstallString"
+	${If} $R0 == ""
+		Goto Done
+	${EndIf}
+	
+	;Check if any other version is installed
+	ReadRegStr $R1 HKLM "${INSTDIR_REG_KEY}" "InstalledVersion"
+	
+	;This check for older versions (pre 1.6.0) without InstalledVersion string will bypass this section
+	${If} $R1 == ""
+	DetailPrint "InstalledVersion string not found, skipping version check"
+		Goto Done
+	${EndIf}
+
+	;Installing same version
+	${If} $R1 S== ${APP_VERSION}
+	MessageBox MB_ICONEXCLAMATION|MB_OKCANCEL "This version of PCSX2 is already installed. Do you want to continue?" IDOK true IDCANCEL false
+	true:
+	DetailPrint "Overwriting current install"
+		Goto Done
+		false:
+		Quit
+		${Else}
+		DetailPrint "Not the same version"
+	${EndIf}
```

Comment:
I don't do NSIS stuff , but it's probably better if it's indented like this:
```
${If} $R1 S== ${APP_VERSION}
	MessageBox MB_ICONEXCLAMATION|MB_OKCANCEL "This version of PCSX2 is already installed. Do you want to continue?" IDOK true IDCANCEL false
	true:
		DetailPrint "Overwriting current install"
		Goto Done
	false:
		Quit
${Else}
	DetailPrint "Not the same version"
${EndIf}
```

From reading the documentation I think you can actually omit `IDOK true` and the true label since you don't actually need to jump in that case.